.devcontainer/devcontainer.json

```diff
@@ -14,11 +14,11 @@
     "SCCACHE_BUCKET": "rapids-sccache-devs",
     "VAULT_HOST": "https://vault.ops.k8s.rapids.ai",
     "HISTFILE": "${containerWorkspaceFolder}/.cache/._bash_history",
-    "DEVCONTAINER_NAME": "cuda12.2-gcc12",
-    "CCCL_CUDA_VERSION": "12.2",
+    "DEVCONTAINER_NAME": "cuda12.3-gcc12",
+    "CCCL_CUDA_VERSION": "12.3",
     "CCCL_HOST_COMPILER": "gcc",
     "CCCL_HOST_COMPILER_VERSION": "12",
-    "CCCL_BUILD_INFIX": "cuda12.2-gcc12",
+    "CCCL_BUILD_INFIX": "cuda12.3-gcc12",
     "CXX": "g++",
     "CUDAHOSTCXX": "g++"
```

Comment:
suggestion: the new containers define these now, so we can drop these defs from the devcontainers.